watchtower/blob/justice_kit.go

```diff
@@ -1,153 +1,288 @@
-	// fund will be deposited. This value is included in the blobs, as
-	// opposed to the session info, such that the sweep addresses can't be
-	// correlated across sessions and/or towers.
-	//
-	// NOTE: This is chosen to be the length of a maximally sized witness
-	// program.
-	SweepAddress []byte
-
-	// RevocationPubKey is the compressed pubkey that guards the revocation
-	// clause of the remote party's to-local output.
-	RevocationPubKey PubKey
-
-	// LocalDelayPubKey is the compressed pubkey in the to-local script of
-	// the remote party, which guards the path where the remote party
-	// claims their commitment output.
-	LocalDelayPubKey PubKey
-
-	// CSVDelay is the relative timelock in the remote party's to-local
-	// output, which the remote party must wait out before sweeping their
-	// commitment output.
-	CSVDelay uint32
-
-	// CommitToLocalSig is a signature under RevocationPubKey using
-	// SIGHASH_ALL.
-	CommitToLocalSig lnwire.Sig
-
-	// CommitToRemotePubKey is the public key in the to-remote output of the
-	// revoked commitment transaction.
-	//
-	// NOTE: This value is only used if it contains a valid compressed
-	// public key.
-	CommitToRemotePubKey PubKey
-
-	// CommitToRemoteSig is a signature under CommitToRemotePubKey using
-	// SIGHASH_ALL.
-	//
-	// NOTE: This value is only used if CommitToRemotePubKey contains a
-	// valid compressed public key.
-	CommitToRemoteSig lnwire.Sig
+// JusticeKit is an interface that describes lé Blob of Justice. An
+// implementation of the JusticeKit contains information required to construct
+// a justice transaction, that sweeps a remote party's revoked commitment
+// transaction. It supports encryption and decryption using chacha20poly1305,
+// allowing the client to encrypt the contents of the blob, and for a
+// watchtower to later decrypt if action must be taken.
+type JusticeKit interface {
+	// ToLocalOutputSpendInfo returns the info required to send the to-local
+	// output. It returns the output pub key script and the witness required
+	// to spend the output.
+	ToLocalOutputSpendInfo() (*txscript.PkScript, wire.TxWitness, error)
+
+	// ToRemoteOutputSpendInfo returns the info required to send the
+	// to-remote output. It returns the output pub key script, the witness
+	// required to spend the output and the sequence to apply.
+	ToRemoteOutputSpendInfo() (*txscript.PkScript, wire.TxWitness, uint32,
+		error)
+
+	// HasCommitToRemoteOutput returns true if the kit does include the
+	// information required to sweep the to-remote output.
+	HasCommitToRemoteOutput() bool
+
+	// AddToLocalSig adds the to-local signature to the kit.
+	AddToLocalSig(sig lnwire.Sig)
+
+	// AddToRemoteSig adds the to-remote signature to the kit.
+	AddToRemoteSig(sig lnwire.Sig)
+
+	// SweepAddress returns the sweep address to be used on the justice tx
+	// output.
+	SweepAddress() []byte
+
+	// PlainTextSize is the size of the encoded-but-unencrypted blob in
+	// bytes.
+	PlainTextSize() int
```

Comment:
do you mean that this should just return the full encoded size? 

Currently the JusticeKit is un-aware of the fact that it gets encrypted - so I think it's better to keep it that way rather than adding the `NonceSize+CiphertextExpansion` to every implementation of the method. 

or do you mean something else?

Reply:
right, it was about adding the other sizes in the interface method, but it makes sense to leave it like it is :+1: